src/YHttpServer.cc

```diff
@@ -50,54 +51,117 @@ int YHttpServer::port_num()
 }
 
 // For security reasons accept the connections only from the localhost
-// by default, allow using INADDR_ANY only when explicitly allowed
-in_addr_t listen_address()
+// by default, allow listening on all interfaces only when explicitly allowed.
+bool remote_access()
 {
     // remote access enabled
     if ((getenv( YUITest_HTTP_REMOTE ) && strcmp(getenv( YUITest_HTTP_REMOTE ), "1") == 0))
     {
         yuiWarning() << "Warning: Enabling remote access!" << std::endl;
-        return INADDR_ANY;
+        return true;
     }
 
-    yuiMilestone() << "Only localhost connections allowed." << std::endl;
     // allow only loop back connection ("http://localhost" only)
-    return htonl(INADDR_LOOPBACK);
+    yuiMilestone() << "Only localhost connections allowed." << std::endl;
+    return false;
+}
+
+// return the IPv4 listen address (localhost or all)
+in_addr_t listen_address_v4(bool remote)
+{
+    return (remote) ? INADDR_ANY : htonl(INADDR_LOOPBACK);
+}
+
+// return the IPv6 listen address (localhost or all)
+struct in6_addr listen_address_v6(bool remote)
+{
+    return (remote) ? in6addr_any : in6addr_loopback;
 }
 
-YHttpServer::YHttpServer() : server(nullptr), redraw(false)
+YHttpServer::YHttpServer() : server_v4(nullptr), server_v6(nullptr), redraw(false)
 {
     _yserver = this;
+
+    // authorization user set?
+    if (const char *user = getenv(YUI_AUTH_USER))
+    {
+        auth_user = user;
+        // unset to not leak to the child processes
+        unsetenv(YUI_AUTH_USER);
```

Comment:
`getenv()` returns a pointer to the environment. Using that pointer is safe as long as the environment, or more exactly that one environment variable, is unchanged. But this `unsetenv()` that follows immediately very likely invalidates the pointer. Are there really still valid data at that point? I'd feel a lot more comfortable copying the content of that environment variable away, say, to a `std::string`.

Reply:
The `auth_user` is a `std::string` (see https://github.com/libyui/libyui-rest-api/pull/2/files#diff-b4b8c37cb1142024f560df026a889317R93), unfortunately the .h file is displayed at the bottom of the Gihub diff...

Reply:
I see. OK then!